maintaining-and-tuning/monitoring.md

```diff
@@ -198,16 +199,189 @@ These instructions assume that GRR server and Prometheus are both up and running
-If you want to disable or remove an alert, go to the dashboard and the corresponding panel, and there you can remove the alerting rule.
+Prometheus supports automatic
+[Service Discovery](https://prometheus.io/docs/prometheus/latest/configuration/configuration/)
+for many types of infrastructure. Depending on your hosting
+setup and size of your GRR installation, this can be an
+improvement over manually hardcoding hostnames in the Prometheus configuration.
 
-1. You can now use the dashboards. The dashboards can give a general overview over the main components of the GRR server and Fleetspeak servers, which can be utilized by the user to monitor different metrics of each component. Examples for such metrics can be found in the [examples above](#example-queries). Remember that the dashboards and alerts are flexible, and can be expanded or modified to adjust to your exact needs. Additional metrics can be used by exploring `http://<host>:<port>/metrics` for each component of GRR server (change the port according to the GRR server component you want) and Fleetspeak servers, and if you'd like to create your own [custom dashboards](https://grafana.com/docs/grafana/latest/getting-started/getting-started/#create-a-dashboard), [panels](https://grafana.com/docs/grafana/latest/panels/add-a-panel/#add-a-panel) and [alerts](https://grafana.com/docs/grafana/latest/alerting/create-alerts/#create-alerts), make sure to go over the corresponding documentation in Grafana.
+### Security Considerations
+A minimal HTTP service, based on
+[prometheus_client](https://github.com/prometheus/client_python/)
+is listening at `Monitoring.http_port` for each GRR component. This HTTP
+service exports read-only metrics under `/metrics` and `/varz` and does
+**not enforce any access control**. People with access to it can read
+aggregated metrics about your GRR installation. With these metrics, facts
+about the number of workers, flow activity, and service health can be
+derived. Make sure to limit access to the port, for example by employing
+a firewall. Furthermore, read [Prometheus Security](https://prometheus.io/docs/operating/security/).
+
+## Monitoring Client Load Stats
+
+In GRR Admin UI you are able to view statistics of an individual GRR
+client. To do this, enter the host information of the client, and on
+the menu, click Advanced and then Client Load Stats. This data is
+gathered from the GRR client itself.
+If you use Grafana dashboards to monitor other GRR server components,
```

Comment:
Let's describe this as *the* way to graph client load stats, not as a "backup" option.